HumanConsoleHandler.java

```diff
@@ -0,0 +1,44 @@
+package mate.academy.handler;
+
+import mate.academy.dao.Dao;
+import mate.academy.di.Inject;
+import mate.academy.di.Injector;
+import mate.academy.model.Human;
+
+import java.util.Scanner;
+
+public class HumanConsoleHandler implements ConsoleHandler{
+
+    @Inject
+    private static Dao humanDao;

```

Comment:
why static ?

Reply:
> why static ?

because this instance is common to all instances of this class

Reply:
it should not be the same for all instances, otherwise why do you need dependency injection?

Reply:
> it should not be the same for all instances, otherwise why do you need dependency injection?

well, let's say we have an application for recording clients and reading them. To have some kind of interface for management, we create a client ConsoleHandler class. Suppose in one place we only need to write, in the other only to read, somewhere else this and that, and somewhere there will be a functional of a class that has not yet been implemented. For all these methods, you need to know if the file has access, if there is access to the operational data, and for example, we will add access to the database. As mentioned earlier, in the first place we record, check whether there is access. In another we read, again we check if there is access. Why, for each such operation, create new instances of dao, all the more it laments the entire logic of the application, besides this, the code was written before me, which in no case can be changed, but can only be extensible. This class is similar to the class of settings, which, as I know, it is customary to make singleton.